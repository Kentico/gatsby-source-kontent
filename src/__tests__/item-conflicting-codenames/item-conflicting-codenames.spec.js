
const { KontentTestHttpService }
= require('@kentico/kontent-test-http-service-js');

const { sourceNodes } = require('../../gatsby-node');

// Project ID 532c0844-5eb3-0033-7d1b-faeea1e7e407
const conflictingCodenamesItemsResponse =
require('./conflictingCodenamesItemsResponse.json');
const fakeTypeResponse =
require('./fakeTypeResponse.json');
const fakeTaxonomiesResponse =
require('../fakeTaxonomiesResponse.json');

describe(`Codenames with conflicting underscores`, () => {
  const fakeHttpServiceConfig = new Map();
  fakeHttpServiceConfig.set(
    /https:\/\/deliver.kontent.ai\/.*\/items/,
    {
      fakeResponseJson: conflictingCodenamesItemsResponse,
      throwError: false,
    });
  fakeHttpServiceConfig.set(
    /https:\/\/deliver.kontent.ai\/.*\/types/,
    {
      fakeResponseJson: fakeTypeResponse,
      throwError: false,
    });

  fakeHttpServiceConfig.set(
    /https:\/\/deliver.kontent.ai\/.*\/taxonomies/,
    {
      fakeResponseJson: fakeTaxonomiesResponse,
      throwError: false,
    });

  const dummyCreateNodeID = jest.fn();
  dummyCreateNodeID.mockImplementation((input) => `dummy-${input}`);

  const createNodeMock = jest.fn();
  const createTypesMock = jest.fn();
  const mockedSchema = { buildObjectType: jest.fn((input) => ({
    data: input,
  }))}; ;

  const actions = {
    actions: {
      createNode: createNodeMock,
      createTypes: createTypesMock,
    },
    createNodeId: dummyCreateNodeID,
    schema: mockedSchema,
  };

  const deliveryClientConfig = {
    projectId: 'dummyProject',
    typeResolvers: [],
    httpService: new KontentTestHttpService(
      fakeHttpServiceConfig,
    ),
  };

  const pluginConfiguration = {
    deliveryClientConfig,
    languageCodenames: ['default'],
  };

  it('passes with no error', async () => {
    await sourceNodes(actions, pluginConfiguration);

    const calls = createNodeMock.mock.calls;
    expect(calls).toMatchSnapshot();
  });
});
