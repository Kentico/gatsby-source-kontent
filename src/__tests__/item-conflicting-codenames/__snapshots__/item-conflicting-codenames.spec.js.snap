// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Codenames with conflicting underscores passes with no error 1`] = `
Array [
  Array [
    Object {
      "children": Array [],
      "id": "dummy-kentico-kontent-taxonomy-personas",
      "internal": Object {
        "contentDigest": "91ece672836106bfc0f186606f20a01d",
        "type": "KontentTaxonomyPersonas",
      },
      "parent": null,
      "system": Object {
        "codename": "personas",
        "id": "f30c7f72-e9ab-8832-2a57-62944a038809",
        "lastModified": "2016-10-20T13:24:00.3200000Z",
        "name": "Personas",
      },
      "terms": Array [
        Object {
          "codename": "coffee_expert",
          "name": "Coffee expert",
          "terms": Array [
            Object {
              "codename": "barista",
              "name": "Barista",
              "terms": Array [],
            },
            Object {
              "codename": "cafe_owner",
              "name": "Cafe owner",
              "terms": Array [],
            },
          ],
        },
        Object {
          "codename": "coffee_enthusiast",
          "name": "Coffee enthusiast",
          "terms": Array [
            Object {
              "codename": "coffee_lover",
              "name": "Coffee lover",
              "terms": Array [],
            },
            Object {
              "codename": "coffee_blogger",
              "name": "Coffee blogger",
              "terms": Array [],
            },
          ],
        },
      ],
      "usedByContentItems___NODE": Array [],
    },
  ],
  Array [
    Object {
      "children": Array [],
      "contentItems___NODE": Array [
        "dummy-kentico-kontent-item-__child-default",
        "dummy-kentico-kontent-item-_child-default",
        "dummy-kentico-kontent-item-chi__ld-default",
        "dummy-kentico-kontent-item-chi_ld-default",
        "dummy-kentico-kontent-item-child-default",
      ],
      "elements": Array [
        Object {
          "codename": "title",
          "name": "Title",
          "options": Array [],
          "type": "text",
        },
      ],
      "id": "dummy-kentico-kontent-type-child",
      "internal": Object {
        "contentDigest": "81f25311660f4dd7fb0117f729b022c8",
        "type": "KontentTypeChild",
      },
      "parent": null,
      "system": Object {
        "codename": "child",
        "id": "7593cb08-f848-42dd-b6df-b8e619025a3c",
        "lastModified": "2020-03-03T09:15:59.4486118Z",
        "name": "Child",
      },
      "usedByContentItems___NODE": Array [],
    },
  ],
  Array [
    Object {
      "children": Array [],
      "contentItems___NODE": Array [
        "dummy-kentico-kontent-item-conflicted_codenames-default",
      ],
      "elements": Array [
        Object {
          "codename": "title",
          "name": "Title",
          "options": Array [],
          "type": "text",
        },
        Object {
          "codename": "children",
          "name": "Children",
          "options": Array [],
          "type": "modular_content",
        },
      ],
      "id": "dummy-kentico-kontent-type-parent",
      "internal": Object {
        "contentDigest": "6a2d9285c3b1c3de788764dd872583df",
        "type": "KontentTypeParent",
      },
      "parent": null,
      "system": Object {
        "codename": "parent",
        "id": "d1cd2f83-60e8-4416-b2e7-d73b1361d3e7",
        "lastModified": "2020-03-03T09:15:37.2966743Z",
        "name": "Parent",
      },
      "usedByContentItems___NODE": Array [],
    },
  ],
  Array [
    Object {
      "children": Array [],
      "contentType___NODE": "dummy-kentico-kontent-type-child",
      "elements": Object {
        "title": Object {
          "name": "Title",
          "type": "text",
          "value": "__child",
        },
      },
      "id": "dummy-kentico-kontent-item-__child-default",
      "internal": Object {
        "contentDigest": "3959787e5d5552d79209516b5fb2e88d",
        "type": "KontentItemChild",
      },
      "otherLanguages___NODE": Array [],
      "parent": null,
      "preferred_language": "default",
      "system": Object {
        "codename": "__child",
        "id": "60fbe92a-a491-43b5-841f-e8c838d7b31f",
        "language": "default",
        "lastModified": "2020-03-03T09:17:43.317Z",
        "name": "__child",
        "sitemapLocations": Array [],
        "type": "child",
      },
      "usedByContentItems___NODE": Array [
        "dummy-kentico-kontent-item-conflicted_codenames-default",
      ],
    },
  ],
  Array [
    Object {
      "children": Array [],
      "contentType___NODE": "dummy-kentico-kontent-type-child",
      "elements": Object {
        "title": Object {
          "name": "Title",
          "type": "text",
          "value": "_child",
        },
      },
      "id": "dummy-kentico-kontent-item-_child-default",
      "internal": Object {
        "contentDigest": "b3ec086bb7d24eb366112f7f75dbe742",
        "type": "KontentItemChild",
      },
      "otherLanguages___NODE": Array [],
      "parent": null,
      "preferred_language": "default",
      "system": Object {
        "codename": "_child",
        "id": "04e1d690-ee87-4791-931c-64ef5343c2cd",
        "language": "default",
        "lastModified": "2020-03-03T09:16:58.873Z",
        "name": "_child",
        "sitemapLocations": Array [],
        "type": "child",
      },
      "usedByContentItems___NODE": Array [
        "dummy-kentico-kontent-item-conflicted_codenames-default",
      ],
    },
  ],
  Array [
    Object {
      "children": Array [],
      "contentType___NODE": "dummy-kentico-kontent-type-child",
      "elements": Object {
        "title": Object {
          "name": "Title",
          "type": "text",
          "value": "chi__ld",
        },
      },
      "id": "dummy-kentico-kontent-item-chi__ld-default",
      "internal": Object {
        "contentDigest": "7a68795a05a97fb8f4717b2b5351df27",
        "type": "KontentItemChild",
      },
      "otherLanguages___NODE": Array [],
      "parent": null,
      "preferred_language": "default",
      "system": Object {
        "codename": "chi__ld",
        "id": "c7a0bf25-0acf-4a35-8a82-48360afc64bb",
        "language": "default",
        "lastModified": "2020-03-03T09:18:52.451Z",
        "name": "chi__ld",
        "sitemapLocations": Array [],
        "type": "child",
      },
      "usedByContentItems___NODE": Array [
        "dummy-kentico-kontent-item-conflicted_codenames-default",
      ],
    },
  ],
  Array [
    Object {
      "children": Array [],
      "contentType___NODE": "dummy-kentico-kontent-type-child",
      "elements": Object {
        "title": Object {
          "name": "Title",
          "type": "text",
          "value": "chi_ld",
        },
      },
      "id": "dummy-kentico-kontent-item-chi_ld-default",
      "internal": Object {
        "contentDigest": "cbac187d9e65b0785c3ba67b24636f16",
        "type": "KontentItemChild",
      },
      "otherLanguages___NODE": Array [],
      "parent": null,
      "preferred_language": "default",
      "system": Object {
        "codename": "chi_ld",
        "id": "71984f0b-fbb1-4f25-b329-095e9aaa6fa0",
        "language": "default",
        "lastModified": "2020-03-03T09:18:16.183Z",
        "name": "chi_ld",
        "sitemapLocations": Array [],
        "type": "child",
      },
      "usedByContentItems___NODE": Array [
        "dummy-kentico-kontent-item-conflicted_codenames-default",
      ],
    },
  ],
  Array [
    Object {
      "children": Array [],
      "contentType___NODE": "dummy-kentico-kontent-type-child",
      "elements": Object {
        "title": Object {
          "name": "Title",
          "type": "text",
          "value": "child",
        },
      },
      "id": "dummy-kentico-kontent-item-child-default",
      "internal": Object {
        "contentDigest": "85fd0a581609e763f24158c40cece0f1",
        "type": "KontentItemChild",
      },
      "otherLanguages___NODE": Array [],
      "parent": null,
      "preferred_language": "default",
      "system": Object {
        "codename": "child",
        "id": "2d995afc-8e73-4d10-bd19-fc2857c19e92",
        "language": "default",
        "lastModified": "2020-03-03T09:17:17.778Z",
        "name": "child",
        "sitemapLocations": Array [],
        "type": "child",
      },
      "usedByContentItems___NODE": Array [
        "dummy-kentico-kontent-item-conflicted_codenames-default",
      ],
    },
  ],
  Array [
    Object {
      "children": Array [],
      "contentType___NODE": "dummy-kentico-kontent-type-parent",
      "elements": Object {
        "children": Object {
          "itemCodenames": Array [
            "child",
            "_child",
            "__child",
            "chi_ld",
            "chi__ld",
          ],
          "linked_items___NODE": Array [
            "dummy-kentico-kontent-item-child-default",
            "dummy-kentico-kontent-item-_child-default",
            "dummy-kentico-kontent-item-__child-default",
            "dummy-kentico-kontent-item-chi_ld-default",
            "dummy-kentico-kontent-item-chi__ld-default",
          ],
          "name": "Children",
          "type": "modular_content",
        },
        "title": Object {
          "name": "Title",
          "type": "text",
          "value": "Conflicted Codenames",
        },
      },
      "id": "dummy-kentico-kontent-item-conflicted_codenames-default",
      "internal": Object {
        "contentDigest": "c1ab8d9a8538814b5777b9d75275f9b2",
        "type": "KontentItemParent",
      },
      "otherLanguages___NODE": Array [],
      "parent": null,
      "preferred_language": "default",
      "system": Object {
        "codename": "conflicted_codenames",
        "id": "c66d1a0b-a9b6-4b45-8e6f-5f30550806f0",
        "language": "default",
        "lastModified": "2020-03-03T09:18:44.142Z",
        "name": "Conflicted Codenames",
        "sitemapLocations": Array [],
        "type": "parent",
      },
      "usedByContentItems___NODE": Array [],
    },
  ],
]
`;
